### DISCUSSION TITLE
How do login scripts work?

### DISCUSSION THREAD

**Discussion by MrIDK22** [Eugeny/tabby#6104] [Q&A]
Hi, 

I've added a test login script with the following values:
- Expect: `prr`
- Send: `echo Testing`

I was expecting that when I ssh to the remote server and send `prr`, it should send `echo Testing`, but it just returns `prr: command not found`. Is this the right use case or am I misunderstanding something?

**Comment by MrIDK22** [comment]
Any updates? (Shameless tag @Eugeny)

**Reply by Eugeny** [reply]
The login scripts work in the same way the `expect` utility does - it listens for specific output from the server and then sends the given input. It's used to e.g. automatically respond to password prompts.

**Reply by MrIDK22** [reply]
Ahh, so it's not setting up a script that runs on login successful. It seems I misunderstood, sorry about that.

Is there any way to include a script that will be automatically executed (I tried # but didn't seem to work) when ssh-ing into the server or setting up snippets / aliases like in Termius?

**Reply by Eugeny** [reply]
`#` should work if it's actually a part of the prompt shown by the server. There's also a 3rd party `quick-cmds` plugin that lets you bind command inputs to keys.

**Reply by MrIDK22** [reply]
Ok, I got it to work after restarting the application. I saw that `quick-cmds` was no longer maintained, but it seems to be a good solution for I need so far. Thanks a lot Eugeny :)

**Comment by gitthangbaby** [comment]
What variables can be used in the command? I'd like to repeat the password for sudo.

**Comment by gitthangbaby** [comment]
What does "Optional" option means? Couldn't figure out logically what it is.

**Comment by and7ey** [comment]
`The login scripts work in the same way the expect utility does - it listens for specific output from the server and then sends the given input. It's used to e.g. automatically respond to password prompts.`

suggest to add it to Wiki - I've spend some time trying to understand what Expect in Tabby UI means
